enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,209 @@
+disable the driver and/or snapshotter, they would be deployed again when the service restarts. Additionally, the LVMS an
+CSI containers required to run LVMS are always packaged with the product and thus will always consume a certain amount
+of storage space devices where MicroShift is deployed.
+
+## Motivation
+
+Not all users may persistent storage provider or volume snapshot capabilities for their MicroShift deployments and
+should be enabled to choose whether to deploy the driver and / or snapshotter. This will afford resource-conscious users
+the opportunity to better tune their MicroShift deployment to their hardware requirements.
+
+### User Stories
+
+- A user deploys MicroShift with the LVMS driver and CSI snapshot controller.
+- A user deploys MicroShift with the LVMS driver only.
+- A user deploys MicroShift with neither the LVMS driver nor the CSI snapshot controller.
+
+### Goals
+
+- Provide the LVMS CSI driver and LVMS CSI snapshot controller as optional components of MicroShift
+- Reuse the optional-installation pattern implemented by MicroShift for the [Multus CNI
+  Plugin](./multus-cni-for-microshift.md).
+- Enable installing the LVMS CSI driver and LVMS CSI snapshot controller to a cluster that was previously deployed
+  without the components.
+
+### Non-Goals
+
+- Generalizing this design to install any other CSI driver.
+- Uninstalling the CSI driver or snapshotter rpms will not uninstall the cluster level components. This is a data safety
+  concern that should be handled by the user.
+- Support installing CSI snapshotter, but not the CSI driver. The snapshotter depends on the CSI driver; it cannot
+  run as a standalone component.
+
+## Proposal
+
+Extract away deployment of the LVMS CSI driver and LVMS CSI snapshot controller from MicroShift logic and provide the
+components as separate, optional rpms. Users should be able to install these rpm simultaneously with core MicroShift
+rpms. It should be possible to install the CSI components to a cluster that was initially deployed without them.
+This should not be a reversible process: uninstalling the rpms should not delete CSI components or cluster APIs from the
+cluster. Doing so endangers users' data and could lead to orphaning of LVM volumes.
+
+### Workflow Description
+
+**_Prerequisites_**
+
+- Host has LVM installed
+
+**_Installation with CSI Driver and Snapshotting_**
+
+1. User determines there is a requirement for persistent storage and volume snapshotting.
+2. User specifies an ostree blueprint which includes the following sections:
+    1. Packages: microshift, microshift-greenboot, microshift-networking, microshift-selinux, microshift-lvms, 
+       microshift-lvms-snapshotting
+    2. File (Optional): lvmd.yaml
+3. User compiles an ostree commit from the blueprint
+4. User deploys the ostree commit to host
+5. MicroShift host boots
+6. MicroShift starts
+7. MicroShift deploys:
+   1. LVMS CSI manifests
+   2. LVMS Snapshot manifests
```

Comment:
I think we have to make it clear that the snapshotting controller is something completely different from LVMS. Thus I also think that installing the snapshot controller should be a separate rpm from lvms. This is also because LVMS can work without snapshot capabilities in the clusters.

Reply:
I agree. That's the intent, but it needs to be stated more explicitly.

Reply:
I've updated several areas in the document to call out the separation of LVMS and the CSI snapshotter.